types/mailgun-js/index.d.ts

```diff
@@ -149,6 +160,23 @@ declare namespace Mailgun {
             address: string,
             callback?: (error: Error, body: validation.ValidateResponse) => void
         ): Promise<validation.ValidateResponse>;
+        validate(
+            address: string,
+            isPrivate: boolean,
+            callback?: (error: Error, body: validation.ValidateResponse) => void
+        ): Promise<validation.ValidateResponse>;
+        validate(
+            address: string,
+            // tslint:disable-next-line unified-signatures
+            opts: { [key: string]: any },
+            callback?: (error: Error, body: validation.ValidateResponse) => void
+        ): Promise<validation.ValidateResponse>;
```

Comment:
Will passing a callback cause this method to return `void` instead of a `Promise`?

Reply:
Good catch! The functions returns void when a callback is passed in.